lib/logger.js

```diff
@@ -121,6 +121,29 @@ var streamConverter = {
 };
 var streamConverterList = Object.keys(streamConverter);
 
+Logger.prototype._addErrorHandler = function() {
+    var self = this;
+    self._logger.on('error', function(err, failedStream) {
+        // If some fatal error occurred in one of the logger streams,
+        // we can't do much, so ignore.
+
+        if (err.code === 'ENOSPC') {
+            // However, if it's a `file` stream, and the error is from
+            // a filled-up disc, attempting to further write to this
+            // stream will leak memory and finally destroy the process.
+            // So remove destroy the stream and remove it.
+            self._logger.streams = self._logger.streams.filter(function(s) {
```

Comment:
> I guess the worst-case scenario here is that workers will re-die after the SIGHUP and the new children will not have the failed stream.

Re-die? Workers don't die because of this any more. When the partition is filled up the error is caught and the appropriate logger is removed. Workers continue to operate normally.